src/test/java/com/pusher/client/channel/impl/PrivateEncryptedChannelImplTest.java

```diff
@@ -0,0 +1,122 @@
+package com.pusher.client.channel.impl;
+
+import com.pusher.client.AuthorizationFailureException;
+import com.pusher.client.Authorizer;
+import com.pusher.client.connection.impl.InternalConnection;
+import com.pusher.client.util.Factory;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Matchers;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PrivateEncryptedChannelImplTest {
```

Comment:
how about all other test cases? in particular how about testing actually receiving event with encrypted data? currently only prepare method is being tested. [analogous test](https://github.com/pusher/pusher-websocket-java/blob/9745b3dc358b2fe3f111ea17b882a8a879c8c049/src/test/java/com/pusher/client/channel/impl/PrivateChannelImplTest.java#L22) has 39 test cases overral. I'm not sure about that inheritance within tests but definitely it feels like we should test more of representative scenarios around the code of `PrivateEncryptedChannelImpl` and how it may affect its inherited behaviour

Reply:
I think we definitely want to test as much of this Class as possible. We have a ticket already to capture this in more detail, so I just wanted to ensure the new pieces I added worked as we expected so as to keep this PR smaller and get feedback sooner.

Reply:
The separate card is about functional testing as there might been a cross dev dependency. This test here is a unit test.

Reply:
👍 have extended from `ChannelImplTest` with overrides for the tests that needed it. This means we're fully covered (it did find a bug - so worth doing!). I've also added two other tests so the PrivateEncryptedChannelImpl is 100% covered 😎